hera_cal/abscal.py

```diff
@@ -632,60 +793,216 @@ def TT_Phi_gain_arr(self):
+        model_pols) = UVData2AbsCalDict(model_files, pop_autos=True, return_meta=True, pol_select=pol_select)
+    antpos = model_antpos
+
+    # iterate over data files
+    gains = []
+    echo("loading data files", type=1, verbose=verbose)
+    for i, dfile in enumerate(data_files):
+
+        # check output filepath
+        if write_calfits:
+            # configure filename
+            if calfits_fname is None:
+                calfits_fname = os.path.basename(dfile) + '.abscal.calfits'
+            if outdir is None:
+                outdir = os.path.dirname(dfile)
+            calfits_fname = os.path.join(outdir, calfits_fname)
+            
+            # check path
+            if os.path.exists(calfits_fname) and overwrite == False:
+                raise IOError("{} exists, not overwriting".format(calfits_fname))
+
+        # load data and configure weights
+        echo("loading {}".format(dfile), type=1, verbose=verbose)
+        (data, data_flags, data_antpos, data_ants, data_freqs, data_times, data_lsts,
+            data_pols) = UVData2AbsCalDict(dfile, pop_autos=True, return_meta=True, pol_select=pol_select, pick_data_ants=False)
+        # get data ants
+        total_data_antpos = copy.deepcopy(data_antpos)
+        data_ants = np.unique(map(lambda k: k[:2], data.keys()))
+        data_antpos = odict(map(lambda k: (k, data_antpos[k]), data_ants))
+        # get wgts
+        wgts = DataContainer(odict(map(lambda k: (k, (~data_flags[k]).astype(np.float)), data_flags.keys())))
+
+        # match redundant baselines
+        if match_red_bls:
+            data = match_red_baselines(data, data_antpos, model, model_antpos, tol=tol, verbose=verbose)
+            antpos = model_antpos
+
+        # interpolate model to match data
+        if interp_model:
+            interp_model, interp_mflags = interp2d_vis(model, model_lsts, model_freqs,
+                                                       data_lsts, data_freqs)
+
+        # reweight according to redundancy
+        if reweight:
+            wgts = mirror_data_to_red_bls(wgts, model_antpos, tol=tol, weights=True)
+
+        # instantiate class
+        AC = AbsCal(interp_model, data, pol_select=pol_select, antpos=antpos, freqs=data_freqs)
+        total_gain_keys = flatten(map(lambda p: map(lambda k: (k, p), total_data_antpos.keys()), AC.gain_pols))
+
+        gain_list = []
+
+        if delay_cal:
+            if alt_gains:
+                raise ValueError("can't run delay_cal when alt_gains is True")
+            AC.delay_lincal(verbose=verbose, time_avg=True)
+            AC.data = apply_gains(AC.data, (AC.ant_dly_gain, AC.ant_dly_phi_gain))
+            gain_list.append(AC.ant_dly_gain)
+            gain_list.append(AC.ant_dly_phi_gain)
+
+        if avg_phs_cal:
+            if delay_cal == False:
+                echo("it is recommended to run a delay_cal before avg_phs_cal", verbose=verbose)
```

Comment:
These might rise to the level of warnings rather than echos. Your call.

Reply:
I don't want to manually restrict the kinds of analysis the user can do, even if to us it seems odd. that seems unneccessarily restrictive.

Reply:
ah sorry, you said warnings--i read errors for some reason.. yeah warnings and echoes seem the same to me, unless the user insists on turning verbose off (which is set to on by default)